File: ClimbLog.Api/ClimbLog.Domain/Models/Entities/Gym.cs
namespace ClimbLog.Domain.Models.Entities;

public class Gym
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int GradeSystemId { get; set; }
    public int ViewBoxXSize { get; set; }
    public int ViewBoxYSize { get; set; }
    public int? StandardGradeSystemId { get; set; }
    public string? Address { get; set; }
    public required string TimeZoneIdentifier { get; set; }
    public required string FloorPlanSvgUrl { get; set; }
    public bool EnableSectors {  get; set; }
    //Navigational Properties
    public List<GymFacility>? GymFacilities { get; set; }
    public List<GymDefaultOpeningHour>? GymDefaultOpeningHours { get; set; }
    public List<GymAdmin>? GymAdmins { get; set; }
    public List<GymCompetition>? GymCompetitions { get; set; }
    public List<Route>? Routes { get; set; }
}

File: climb-log-flutter/lib/climasys/api/models/gym.dart
class Gym {
  final int id;
  final String name;
  final int gradeSystemId;
  final int? standardGradeSystemId;
  final String floorPlanImgUrl;
  final int viewBoxXSize;
  final int viewBoxYSize;
  final bool enableSectors;

  Gym({
    required this.id,
    required this.name,
    required this.gradeSystemId,
    required this.standardGradeSystemId,
    required this.floorPlanImgUrl,
    required this.viewBoxXSize,
    required this.viewBoxYSize,
    required this.enableSectors
  });

  factory Gym.fromJson(Map<String, dynamic> json) {
    return Gym(
      id: json['id'],
      name: json['name'],
      gradeSystemId: json['gradeSystemId'],
      standardGradeSystemId: json['standardGradeSystemId'],
      floorPlanImgUrl: json['floorPlanSvgUrl'],
      viewBoxXSize: json['viewBoxXSize'],
      viewBoxYSize: json['viewBoxYSize'],
      enableSectors: json['enableSectors']
    );
  }
}